drivers/net/mlx5/mlx5_vdpa.c

```diff
@@ -181,27 +187,76 @@ create_rq(struct vdpa_priv *priv, uint16_t qsize, uint16_t idx)
 {
 	uint32_t in[MLX5_ST_SZ_DW(create_rq_in)] = {0};
 	uint32_t out[MLX5_ST_SZ_DW(create_rq_out)] = {0};
+	struct mlx5dv_devx_umem *buf_umem = NULL;
+	struct mlx5dv_devx_umem *dbr_umem = NULL;
 	struct mlx5dv_devx_obj *rq_obj = NULL;
+	void *rq_buf = NULL;
+	void *dbrec = NULL;
 	void *rqc = NULL;
 	void *wq = NULL;
+	int wq_sz;
 
 	MLX5_SET(create_rq_in, in, opcode, MLX5_CMD_OP_CREATE_RQ);
 	rqc = MLX5_ADDR_OF(create_rq_in, in, ctx);
 	MLX5_SET(rqc, rqc, cqn, priv->cqn);
 	wq = MLX5_ADDR_OF(rqc, rqc, wq);
 	/* TODO(idos): Check log_wq_size according to device CAP */
+	wq_sz = qsize * (1 << MLX5_VDPA_LOG_RQ_STRIDE);
+	MLX5_SET(wq, wq, log_wq_stride, MLX5_VDPA_LOG_RQ_STRIDE);
 	MLX5_SET(wq, wq, log_wq_sz, rte_log2_u32(qsize));
 	MLX5_SET(wq, wq, pd, priv->pdn);
+	rq_buf = rte_malloc(__func__, wq_sz, 0);
+	if (!rq_buf) {
+		DRV_LOG(ERR, "Error allocating memory for RQ Buffer");
+		return -1;
+	}
+	dbrec = rte_malloc(__func__, MLX5_VDPA_DBR_RQ_SIZE, 0);
+	if (!dbrec) {
+		DRV_LOG(ERR, "Error allocating memory for RQ DB record");
+		rte_free(rq_buf);
+		return -1;
+	}
+	buf_umem = mlx5_glue->dv_devx_umem_reg(priv->ctx, rq_buf, wq_sz,
+					       IBV_ACCESS_LOCAL_WRITE);
```

Comment:
what about read access?

Reply:
AFAIU there is no IBV_ACCESS_LOCAL_READ (always allowed).
There is a REMOTE_READ for RDMA verbs.
Right?